crates/sui-indexer-alt-graphql/src/api/types/object.rs

```diff
@@ -226,31 +226,29 @@ impl Object {
     }
 
     /// The version of this object that this content comes from.
-    pub(crate) async fn version(&self, ctx: &Context<'_>) -> Result<Option<UInt53>, RpcError> {
+    pub(crate) async fn version(&self, ctx: &Context<'_>) -> Option<Result<UInt53, RpcError>> {
         if let Some((version, _)) = self.version_digest {
-            return Ok(Some(version.into()));
+            return Some(Ok(version.into()));
         }
 
         // Fall back to loading from contents
-        let Some(contents) = self.contents(ctx).await?.as_ref() else {
-            return Ok(None);
-        };
-
-        Ok(Some(contents.version().into()))
+        self.contents(ctx)
+            .await
+            .map(|opt| opt.as_ref().map(|c| c.version().into()))
+            .transpose()
```

Comment:
nit: this pattern is a little hard to follow, I wonder if it would be neater to transpose earlier? Something like:

```rust
let contents = match self.contents(ctx).await.transpose()? {
    Ok(contents) => contents,
    Err(e) => return Some(Err(e)),
};

Some(Ok(contents.version().into()))
```